helm-plugins/snapshot/main.go

```diff
@@ -0,0 +1,244 @@
+	filename := filepath.Base(uri)
+	return strings.TrimSuffix(filename, filepath.Ext(filename))
+}
+
+// CreateMesheryDesign creates a Meshery Design via API
+func CreateMesheryDesign(config *Config, uri, name, email string) (string, error) {
+	payload := MesheryDesignPayload{
+		Save:  true,
+		URL:   uri,
+		Name:  name,
+		Email: email,
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return "", err
+	}
+	sourceType := "Helm Chart"
+	req, err := http.NewRequest("POST", fmt.Sprintf("https://playground.meshery.io/api/pattern/%s", sourceType), bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set("Cookie", config.Cookie)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept-Encoding", "gzip, deflate, br, zstd")
+	req.Header.Set("Accept-Language", "en-GB,en-US;q=0.9,en;q=0.8")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return "", fmt.Errorf("failed to create meshery design. response: %s", string(body))
+	}
+
+	// Expecting a JSON array in the response
+	var result []map[string]interface{}
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return "", err
+	}
+
+	if len(result) > 0 {
+		if id, ok := result[0]["id"].(string); ok {
+			return id, nil
+		}
+	}
+
+	return "", errors.New("invalid response from meshery API")
+}
+
+// GenerateSnapshot triggers the snapshot generation
+func GenerateSnapshot(config *Config, designID, chartURI, assetLocation string) error {
+
+	payload := map[string]interface{}{
+		"Owner":       config.Owner,
+		"Repo":        config.Repo,
+		"Workflow":    config.Repo,
+		"Branch":      config.Branch,
+		"GithubToken": config.GithubToken,
+		"Payload": map[string]string{
+			"application_type": "Helm Chart",
+			"designID":         designID,
+			"assetLocation":    assetLocation,
+		},
+	}
+
+	// Marshal the payload into JSON
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	// Create the POST request
+	req, err := http.NewRequest(
+		"POST",
+		fmt.Sprintf("%s/api/integrations/trigger/workflow", config.MesheryCloudBaseUrl),
+		bytes.NewBuffer(payloadBytes),
+	)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Cookie", config.Cookie)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+config.GithubToken) // Use the GitHub token for authentication
+
+	// Send the request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Check the response status
+	if resp.StatusCode != http.StatusNoContent {
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("failed to dispatch workflow. response: %s", string(body))
+	}
+
+	return nil
+}
+
+func main() {
+	// Initialize the logger
+	var err error
```

Comment:
an ideal world, all Meshery ecosystem components, use MeshKit errors, and provide error codes with that are captured and  incorporated into the error code reference in Meshery Docs.